JASP-Engine/JASP/R/binomialtest.R

```diff
@@ -15,224 +15,315 @@
 
-    for (var in variables) {
-      d <- dataset[[.v(var)]]
-      d <- d[!is.na(d)]
-      levels <- levels(d)
-      n <- length(d)
-
-      # !! Test each level in each variable against test value !!
-      for (lev in levels) {
-        counts <- sum(d == lev)
-        prop <- counts/n
-        r <- stats::binom.test(counts, n, p = options$testValue,
-                               alternative = hyp,
-                               conf.level = options$confidenceIntervalInterval)
-
-        p <- r$p.value
-        vovk <- .VovkSellkeMPR(p)
-        cilo <- r$conf.int[1]
-        ciup <- r$conf.int[2]
-
-        if (p == FALSE) {
-          p <- 0
-        } else if (p == TRUE) {
-          p <- 1
-        }
-
-        binomResults <- rbind(binomResults, c(var, lev, counts, n, prop, p, vovk, cilo, ciup))
-      }
+  # Create table
+  binomialTable <- createJaspTable(title = "Binomial Test")
+  jaspResults[["binomialTable"]] <- binomialTable
+  binomialTable$dependOnOptions(c("variables", "testValue", "hypothesis", "confidenceInterval",
+                                  "confidenceIntervalInterval", "VovkSellkeMPR"))
+
+  binomialTable$showSpecifiedColumnsOnly <- TRUE
+
+  # Add columns to table
+  binomialTable$addColumnInfo(name = "variable",   title = "Variable",   type = "string", combine = TRUE)
+  binomialTable$addColumnInfo(name = "level",      title = "Level",      type = "string")
+  binomialTable$addColumnInfo(name = "counts",     title = "Counts",     type = "integer")
+  binomialTable$addColumnInfo(name = "total",      title = "Total",      type = "integer")
+  binomialTable$addColumnInfo(name = "proportion", title = "Proportion", type = "number", format = "sf:4")
+  binomialTable$addColumnInfo(name = "p",          title = "p",          type = "number", format = "dp:3;p:.001")
+
+  if (options$VovkSellkeMPR) {
+    binomialTable$addColumnInfo(name = "VovkSellkeMPR", title = "VS-MPR\u002A", type = "number", format = "sf:4")
+  }
+
+  if (options$confidenceInterval) {
+    binomialTable$addColumnInfo(name = "lowerCI", title = "Lower", type = "number", format = "sf:4",
+      overtitle = paste0(100 * options$confidenceIntervalInterval, "% CI for Proportion"))
+    binomialTable$addColumnInfo(name = "upperCI", title = "Upper", type = "number", format = "sf:4",
+      overtitle = paste0(100 * options$confidenceIntervalInterval, "% CI for Proportion"))
+  }
+
+  for (variable in options$variables) {
+    for (level in binomResults$spec$levels[[variable]]) {
+      row <- binomResults[[variable]][[level]][1:9]
+      binomialTable$addRows(row, rowNames = paste0(variable, " - ", level))
     }
+  }
 
+  # Add footnote: VovkSellkeMPR
+  if (options$VovkSellkeMPR) {
+    binomialTable$addFootnote(message = .messages("footnote", "VovkSellkeMPR"), symbol = "\u002A")
   }
 
-  return(binomResults)
+  # Add footnote: Alternative hypothesis
+  if (binomResults$spec$hypothesisRec == "two.sided") {
```

Comment:
Typically, I find a switch statement more readable:
```suggestion
  message <- switch(binomResults$spec$hypothesisRec,
    "two.sided" = .messages("footnote", "binomNeq",     value = options$testValue),
    "greater"   = .messages("footnote", "binomGreater", value = options$testValue),
    "less"      = .messages("footnote", "binomLess",    value = options$testValue)
  )
  binomialTable$addFootnote(message = message, symbol = "<em>Note.</em>")
```
When to use this is pretty much up to the developer, but since this should be an example analysis I figured I'd bring it up.

Reply:
great suggestion